### PR TITLE
Move to using jwt encryption with shared key

### PR DESCRIPTION
**Pull request by iwootten**
NB: Hold off merging this until @warrenbailey's PR https://github.com/ONSdigital/eq-survey-runner/pull/126 has been merged into eq and we are post demo.

This is pretty much lifting the majority of the keys/code from Warrens work. Posie will now only respond with json.

### CODE REVIEW

File: encrypter.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+
+import settings
+import jwt
+import os
+import base64
+
+KID = 'SDE'
```
Comment (warrenbailey) [inline encrypter.py]: I'm assuming this class is just for testing?

Comment (iwootten) [inline encrypter.py]: Yup, probably should just move this inside my testservice class, as you had.

File: settings.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    :return:
+    """
+    logger.debug("Opening file %", key_name)
+    key = open(key_name, 'r')
+    contents = key.read()
+    logger.debug("Key is %s", contents)
+    return contents
+
+EQ_JWT_LEEWAY_IN_SECONDS = 120
+
```
Comment (warrenbailey) [inline settings.py]: These seem backwards.

File: settings.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+PUBLIC_KEY = get_key(os.getenv('PUBLIC_KEY', "./jwt-test-keys/sdx-public.pem"))
```
Comment (warrenbailey) [inline settings.py]: We're going to encrypt the message with your (SDX) public key, so you need to use your private key to decrypt.

File: settings.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+PRIVATE_SIGNING_KEY = get_key(os.getenv('PRIVATE_SIGNING_KEY', "./jwt-test-keys/sdx-private.pem"))
```
Comment (warrenbailey) [inline settings.py]: But we'll sign it with our (SR) private key, so you'll need our public key to check the signing?

File: decrypter.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            backend=backend
+        )
+
+    def _to_bytes(self, bytes_or_str):
+        if isinstance(bytes_or_str, str):
+            value = bytes_or_str.encode()
+        else:
+            value = bytes_or_str
+        return value
+
```
Comment (warrenbailey) [inline decrypter.py]: We probably should check the KID value in the header (sorry that's not in my decrypt code)

**Comment by warrenbailey** [comment]
Looks good, just need to rename the keys

File: settings.py (hunk 4 above)
Comment (iwootten) [inline settings.py]: Should I just be removing the "SIGNING_" from these vars then?

**Comment by warrenbailey** [comment]
The key names are better now, lets give it a go :+1:

**Comment by iwootten** [comment]
Going to take @warrenbailey's feedback as a go for launch!

**Pull request merged**